### PR TITLE
refactor(backend): use LogError on contributionLinkResolver

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

fix logger on contributionLinkResolver

### Issues

- contributes to #2487 
- contributes to #1870 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/ContributionLinkResolver.ts [hunk 1]
```diff
@@ -39,35 +40,30 @@ export class ContributionLinkResolver {
     }: ContributionLinkArgs,
   ): Promise<ContributionLink> {
     isStartEndDateValid(validFrom, validTo)
+    // TODO: this should be enforced by the schema.
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: It is. If you call the API without `name`, you get a schema error.
The test is the same as `name.length === 0` and there for obsolete as we test that length must be greater than `x`

File: backend/src/graphql/resolver/ContributionLinkResolver.ts [hunk 2]
```diff
@@ -39,35 +40,30 @@ export class ContributionLinkResolver {
...
-      const msg = `The value of 'name' with a length of ${name.length} did not fulfill the requested bounderies min=${CONTRIBUTIONLINK_NAME_MIN_CHARS} and max=${CONTRIBUTIONLINK_NAME_MAX_CHARS}`
-      logger.error(`${msg}`)
-      throw new Error(`${msg}`)
+    if (name.length < CONTRIBUTIONLINK_NAME_MIN_CHARS) {
+      throw new LogError('The value of name is too short', name.length)
     }
+    if (name.length > CONTRIBUTIONLINK_NAME_MAX_CHARS) {
+      throw new LogError('The value of name is too long', name.length)
+    }
+    // TODO: this should be enforced by the schema.
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionLinkResolver.ts:53]: It is. See above. `'' ? true : false  -> false`

**Review by Mogge — CHANGES_REQUESTED** [review]
Please kick those to dos.

File: backend/src/graphql/resolver/ContributionLinkResolver.ts (hunk 1 above)
Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: image

the whole thing requires refactoring - lets talk about solutions

Comment (Mogge) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: Yes, we can do this. My point is, that the test `(!name)` can only be true if the name is an empty string and therefore useless, as we test later for a minimum length

Comment (Mogge) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: How can we localize error messages thrown by this validator?

Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: > How can we localize error messages thrown by this validator?

We do not localize Error messages - but I am sure there is a way.

Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: I see your point and will remove the check & test

Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionLinkResolver.ts]: Solved with your suggestion and created Issue #2672 for the validator part.

**Review by Mogge — APPROVED** [review]

**Pull request merged**